### ISSUE TITLE
Unable to Resolve Dependency

### ISSUE THREAD

**Issue by Fosol** [aspnet/Home#804]
I've created a very simple class library.  It builds ok if I don't "Produce outputs on build".  If I tell it to "Produce outputs on build" and thus make a NuGet package it fails and gives me the following error;

1>    Unable to resolve dependency System.Linq 4.0.0-beta-23019
1>    Unable to resolve dependency System.Threading 4.0.10-beta-23019
1>    Unable to resolve dependency System.Runtime 4.0.10-beta-23019
1>    Unable to resolve dependency Microsoft.CSharp 4.0.0-beta-23019

I have all of those NuGet packages and versions.  How do I debug this?

**Comment by Fosol** [comment]
I had to right click on the References and "Restore Packages"...

I'm having a horrible experience so far.  I feel well over my head.

**Comment by davidfowl** [comment]
@Fosol sorry to hear that. Do you have anymore feedback?

**Comment by Fosol** [comment]
First I love the concepts!  Secondly I'm hating the versioning/package/namespace nightmare.

I keep running in an issue where DNX doesn't know what assembly to use/include.  The build works, but runtime fails.

The type 'List<T>' exists in both 'System.Collections, Version=4.0.10.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' and 'mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089'

As far as I can tell this above error is due to DNX and nothing to do with my code.  But I know somewhere in some file there is a reference to both assemblies.  I just can't find it to remove it.

**Comment by Fosol** [comment]
I had to manually go through all my shared projects and edit their project.lock.json files to remove System.Collections.  I had to do this even though I had no references to System.Collections in the project.json files.

This is the kind of stuff that is ruining my experience.  I spend hours debugging something that I would have never come across before.  I use to simply add references to the assemblies I needed and voila done.  Now I have to debug what assemblies are getting pulled in from multiple packages and determine if I have to manually rip them out somehow.
